internal/app/machined/pkg/controllers/hardware/system.go

```diff
@@ -58,59 +74,83 @@ func (ctrl *SystemInfoController) Outputs() []controller.Output {
 //
 //nolint:gocyclo
 func (ctrl *SystemInfoController) Run(ctx context.Context, r controller.Runtime, logger *zap.Logger) error {
-	select {
-	case <-ctx.Done():
-		return nil
-	case <-r.EventCh():
-	}
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-r.EventCh():
+		}
 
-	// smbios info is not available inside container, so skip the controller
-	if ctrl.V1Alpha1Mode == runtimetalos.ModeContainer {
-		return nil
-	}
-	// controller runs only once
-	if ctrl.SMBIOS == nil {
-		s, err := pkgSMBIOS.GetSMBIOSInfo()
+		_, err := safe.ReaderGetByID[*runtime.MetaLoaded](ctx, r, runtime.MetaLoadedID)
 		if err != nil {
-			return err
+			if state.IsNotFoundError(err) {
+				continue
+			}
+
+			return fmt.Errorf("error getting meta loaded resource: %w", err)
 		}
 
-		ctrl.SMBIOS = s
-	}
+		// smbios info is not available inside container, so skip the controller
```

Comment:
we could probably keep it at the top of the function, as in container mode we skip it completely...